src/crimson/os/seastore/cache.h

```diff
@@ -489,17 +494,20 @@ class Cache {
    */
   template <typename T>
   TCachedExtentRef<T> alloc_new_extent(
-    Transaction &t,       ///< [in, out] current transaction
-    seastore_off_t length, ///< [in] length
-    bool delayed = false  ///< [in] whether the paddr allocation of extent is delayed
+    Transaction &t,         ///< [in, out] current transaction
+    seastore_off_t length,  ///< [in] length
+    placement_hint_t hint = placement_hint_t::HOT
   ) {
     LOG_PREFIX(Cache::alloc_new_extent);
-    SUBDEBUGT(seastore_cache, "allocate {} {}B, delay={}",
-              t, T::TYPE, length, delayed);
-    auto ret = CachedExtent::make_cached_extent_ref<T>(
-      alloc_cache_buf(length));
-    t.add_fresh_extent(ret, delayed);
+    SUBTRACET(seastore_cache, "allocate {} {}B, hint={}",
+              t, T::TYPE, length, hint);
+    auto result = p_epm->alloc_new_extent(t, T::TYPE, length, hint);
+    auto ret = CachedExtent::make_cached_extent_ref<T>(std::move(result.bp));
+    ret->set_paddr(result.paddr);
```

Comment:
Changes here make sense to me. So, paddr is determined after alloc_new_extent() if backend type is either pmem or rbm?

Reply:
I suppose RBM can expose its allocator to add to EPM, with the paddr determined inside `ExtentPlacementManager::alloc_new_extent()` according to the hint. There are still works to abstract segment specific logic out of the common components, though.

Reply:
Right. So, paddr here does not need to be a record relative address if the backend type is either pmem or rbm? I wonder when the physical allocation occurs---either here or before submitting the record. (sorry, this is not related to this PR).

Reply:
> paddr here does not need to be a record relative address if the backend type is either pmem or rbm?

Right, this can be the place to assign the absolute physical address if it is possible to decide synchronously. Record-relative is another decision that the extent is allocated in-line with the journal Record, so that paddr will be know when the transaction is submitted.